lib/models/device.coffee

```diff
@@ -111,20 +110,20 @@ getDeviceModel = (deps, opts) ->
 	# @memberof resin.models.device
 	#
 	# @param {Object} options - options
-	# @param {Number} options.appId - Application id
-	# @param {Number} options.deviceId - Device id
+	# @param {String} options.deviceUuid - Device uuid
 	#
 	# @returns {String} - Dashboard URL for the specific device
-	# @throws Exception if either appId or deviceId are empty
+	# @throws Exception if the deviceUuid is empty
 	#
 	# @example
-	# dashboardDeviceUrl = resin.models.device.getDashboardUrl({ appId: 123, deviceId: 456 })
+	# dashboardDeviceUrl = resin.models.device.getDashboardUrl({ deviceUuid: 'a44b544b8cc24d11b036c659dfeaccd8' })
```

Comment:
Since it's just one argument, we could make this match all the other methods, and take a uuidOrId parameter.

Reply:
Do you mean to make it just accept uuid as a parameter?
Adding the Id as well would require to change the method to return a promise, since it will have to make a request to find the uuid in that case.
Is changing the return type to a promise something that we are fine with?

Reply:
That is what I mean, but I hadn't thought about the promise return type part.

I'm on the fence, but I think it's probably still the right choice? Makes it consistent with everything else here, and that's definitely quite valuable from a dev UX point of view. What do you think?

Reply:
Tbh I don't really like the idea of turning a previously sync function to an async one.
I'm thinking that a user with no under the hood insights could write the following:
```js
sdk.model.device.getAllByApplication(12345)
.map(device => {
    sdk.model.device.getDashboardUrl(device.id)
})
```
and that would do N additional requests for data that are already downloaded.
What do you think?

Reply:
Moreover, shouldn't we start hiding IDs from our public APIs as well?

Reply:
> Moreover, shouldn't we start hiding IDs from our public APIs as well?

This is a good point. I don't think we want to strip them immediately, but we should start gently moving in that direction. I'm sold: let's go for a single UUID argument, and keep it sync :+1: